tests/std/tests/P0896R4_views_filter_death/test.cpp

```diff
@@ -13,19 +13,43 @@ using namespace std;
 
 static int some_ints[] = {0, 1, 2, 3};
 
-[[maybe_unused]] constexpr auto lambda = [x = 42](int) { return x == 42; };
-using FV                               = decltype(ranges::filter_view{some_ints, lambda});
+struct test_predicate {
+    struct tag {};
+
+    test_predicate() = default;
+    test_predicate(const test_predicate&) {
+        throw tag{};
+    }
+    test_predicate& operator=(const test_predicate&) = delete;
+
+    constexpr bool operator()(int i) const {
+        return i == 42;
+    }
+};
+
+auto with_no_predicate() {
+    using V = decltype(ranges::filter_view{some_ints, test_predicate{}});
+    V r{some_ints, {}};
+    try {
+        r = V{some_ints, {}};
+    } catch (const test_predicate::tag&) {
+    }
+    return r;
+}
 
 void test_view_predicate() {
-    FV r;
-    (void) r.pred(); // value-initialized filter_view has no predicate
+    auto r = with_no_predicate();
+    (void) r.pred(); // filter_view has no predicate
 }
 
 void test_view_begin() {
-    FV r;
+    auto r = with_no_predicate();
     (void) r.begin(); // N4861 [range.filter.view]/3 forbids calling begin on a filter_view that holds no predicate
 }
 
+constexpr auto lambda = [x = 42](int) { return x == 42; };
+using FV              = decltype(ranges::filter_view{some_ints, lambda});

```

Comment:
was this pattern inspired by a compiler bug or something?

Reply:
No, it's a just more concise than `ranges::filter_view<views::all_t<decltype((some_ints))>, decltype(lambda)>`, while still avoiding writing the array bound for `some_ints` directly (which is fragile) and avoiding the need to explain why there are two sets of parens in the first decltype and one in the second.

It's also implicitly testing the explicit deduction guide for `filter_view`.

Reply:
ah, the explicit deduction guide test is what confused me, since the lambda is quite odd for "return true"